### ISSUE TITLE
underscore conversion for italics broken in pandoc-2.4

### ISSUE THREAD

**Issue by ousia** [jgm/pandoc#5053]
[`pandoc-2.4`](http://pandoc.org/try/?text=%E2%80%94_legibility_%E2%80%94&from=markdown&to=html5) converts

``` markdown
—_legibility_—
```

into

``` html
<p>—_legibility_—</p>
```

`pandoc-2.3` parsed it as:

``` html
<p>—<em>legibility</em>—</p>
```

I’m afraid it might be a bug.

**Comment by jgm** [comment]
My guess is that this is due to commit 9b0bd4ec6f5c9125efb3e36232e6d1f6ac08a728

**Comment by jgm** [comment]
Note: commit 9b0bd4ec6f5c9125efb3e36232e6d1f6ac08a728 made possible the fix to #4635, so simply reverting it breaks that test.

The whole `lastStrPos` thing is a kludge due to not being able to look backwards in the token stream when parsing.  #4635 relies on `(`and other symbols triggering an update to `lastStrPos`, but the parser for `_` emphasis apparently relied on `lastStrPos` only being updated by regular non-symbol strings.

It would be nice to find a cleaner overall solution.  But a simple fix for this would also be welcome.

**Comment by bpj** [comment]
Workaround:

    —[_legibility_]{}—

outputs as

    <p>—<span><em>legibility</em></span>—</p>

Tip: add a dummy class so that you can remove the hack with a filter once the bug gets fixed!

**Comment by agusmba** [comment]
Another workaround would be to switch the order, wouldn't it?

    _—legibility—_

**Comment by ousia** [comment]
@bpj, another workaround would be to use the `<em>` tag.

@agusmba, I’m afraid the reply is straightforward:

![no-1](https://user-images.githubusercontent.com/203767/48365406-22a78480-e6ab-11e8-96b5-29c119732f51.png)

The glyph for the em-dash for roman and italic fonts isn’t the same (at least, in TeX Gyre Pagella).

**Comment by agusmba** [comment]
Ah, I didn't zoom all the way in to see the difference.
For a moment I thought you were shouting back!
Sorry for the noise.

**Comment by ousia** [comment]
@agusmba, I only intended to show the difference between the roman and italic glyphs (it was the best way it came to my mind).

**Comment by jgm** [comment]
Fixed by edc651059ee617cf36b511de080a646d2e6513a4
